### PR TITLE
feat: add Nano Pi R4S support

### PR DESCRIPTION
**Pull request by MDr164**
# Pull Request

## What? (description)

This commit adds initial support for the Nano Pi R4S from Friendlyelec. 

## Why? (reasoning)

This device is a networking focused rk3399 based SBC with two 1G ethernet interfaces, making it perfect for edge or SOHO deployments.

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [x] you ran unit-tests (`make unit-tests`)

Requires https://github.com/siderolabs/pkgs/pull/554 to be merged

### CODE REVIEW

**Comment by MDr164** [comment]
Hi, I'm having trouble running the build system locally. Docker complains about unknown manifests so the build is never triggered on my end. Additionally I'll need a version of the pkgs repo with a not yet upstreamed commit, not sure how to realize that with the build system either yet.

**Comment by frezbo** [comment]
> Hi, I'm having trouble running the build system locally. Docker complains about unknown manifests so the build is never triggered on my end. Additionally I'll need a version of the pkgs repo with a not yet upstreamed commit, not sure how to realize that with the build system either yet.

so you need a buildx builder created from the output of `make help` (on talos side)

wrt to `pkgs` you can build you package by doing something like `make u-boot PLATFORM=linux/arm64 REGISTRY=<your registry> USERNAME=<your registry prefix> PUSH=true`

once done, on talos side edit the Dockerfile and for u-boot section directly reference the image you built previously, once that's done you could do `make installer REGISTRY=<your registry> USERNAME=<your registry prefix> PUSH=true` and then `make sbc-<new sbc name> REGISTRY=<your registry> USERNAME=<your registry prefix> PUSH=true`

**Comment by frezbo** [comment]
if you have tested the pkgs build and confirmed the generated image works, I can merge the pkgs PR and that should help you here

**Comment by frezbo** [comment]
> if you have tested the pkgs build and confirmed the generated image works, I can merge the pkgs PR and that should help you here

the pkgs PR is merged

**Comment by MDr164** [comment]
Thanks for merging that already. Didn't find time to test again yesterday but will do so today. Though my build machine takes quite long for building so might take a bit of time.

**Comment by MDr164** [comment]
Tested it and everything works and the board boots into Talos! Attached is the bootlog from the board.

[talos-r4s.log](https://github.com/siderolabs/talos/files/9373708/talos-r4s.log)

**Review by frezbo — APPROVED** [review]
:cool:

**Comment by frezbo** [comment]
note to self: update the support matrix once 1.2 is cut

**Comment by MDr164** [comment]
Looks like the CI is failing because I bumped the pkgs version tag which included the u-boot update. Anything I can do about that on my end?

**Comment by frezbo** [comment]
Will merge once we cut the 1.3 docs

File: website/content/v1.2/talos-guides/install/single-board-computers/nanopi_r4s.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+curl -Lo /usr/local/bin/talosctl https://github.com/siderolabs/talos/releases/download/{{< release >}}/talosctl-$(uname -s | tr "[:upper:]" "[:lower:]")-amd64
+chmod +x /usr/local/bin/talosctl
+```
+
+## Download the Image
+
+Download the image and decompress it:
+
+```bash
+curl -LO https://github.com/siderolabs/talos/releases/download/{{< release >}}/metal-rockpi_4-arm64.img.xz
```
Comment (rgl) [inline website/content/v1.2/talos-guides/install/single-board-computers/nanopi_r4s.md:27]: this still mentions "rockpi_4", is that intended?

Comment (MDr164) [reply website/content/v1.2/talos-guides/install/single-board-computers/nanopi_r4s.md:27]: Good thing you caught the typo, no that's wrong and I'll fix and rebase the PR then. The generated image is being called metal-nanopi_r4s-arm64.img.xz of course.

**Review by frezbo — APPROVED** [review]

**Review by smira — APPROVED** [review]

**Pull request merged**